File: 1980-find-unique-binary-string/1980-find-unique-binary-string.java
class Solution {
    String ans=""; int n;
    HashSet<String> u=new HashSet<String>();
    boolean ok=false;
    public void helper(int idx,String tmp){
        if(ok) return;
        if(idx==n){
            if(!u.contains(tmp)){
                ok=true; ans=tmp;
            }
            return;
        }
        helper(idx+1,tmp+"0");
        helper(idx+1,tmp+"1");
    }
    public String findDifferentBinaryString(String[] nums) {
        n=nums.length;
        for(String x : nums) u.add(x);
        String tmp="";
        helper(0,tmp);
        return ans;
    }
}

File: 1980-find-unique-binary-string/1980-find-unique-binary-string.cpp
class Solution {
public:
    string ans=""; bool ok=0;
    unordered_set<string> u; int n=0;
    void helper(int idx,string tmp){
        if(ok) return;
        if(idx==n){
            if(u.find(tmp)==u.end()){
                ans=tmp; ok=1;
            }
            return;
        }
        helper(idx+1,tmp+'0');
        helper(idx+1,tmp+'1');
    }
    string findDifferentBinaryString(vector<string>& nums) {
        n=nums.size();
        for(auto x : nums) u.insert(x);
        string tmp="";
        helper(0,tmp);
        return ans;
    }
};